pkg/policy/groups/helpers.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2017-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groups
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/k8s"
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	clientset "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/policy/api"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	parentCNP   = "parentCNP"
+	cnpKindKey  = "CNPKind"
+	cnpKindName = "derivative"
+)
+
+var (
+	globalK8sClient *clientset.Clientset
+	k8sMutex        = lock.Mutex{}
+)
+
+// createDerivativeCNP will return a new CNP based on the given rule.
+func createDerivativeCNP(cnp *cilium_v2.CiliumNetworkPolicy) (*cilium_v2.CiliumNetworkPolicy, error) {
+	derivativeCNP := cnp.DeepCopy()
```

Comment:
If you are deleting the Spec and Specs, why are you creating a DeepCopy at all?

Reply:
labels, and objectmeta.